### PR TITLE
[Merged by Bors] - feat: port Data.Polynomial.Module

### PR DESCRIPTION
**Pull request by wrenna-robson**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by linesthatinterlace** [comment]
I've done as much as I can easily fix here, but I'm having issues with the definition of `comp` that I can't get past. Would appreciate some help.

**Comment by linesthatinterlace** [comment]
Thanks for the comp definition @int-y1. I have fixed the remaining proofs, fixed documentation and checked lints, and this now requires review.

File: Mathlib/Data/Polynomial/Module.lean [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+open Polynomial
+
+open Polynomial BigOperators
+
+
+
+-- porting note: commenting out include
+--include R
+
```
Comment (semorrison) [inline Mathlib/Data/Polynomial/Module.lean:34]:
```suggestion
```
Please just remove `include` and `omit` entirely, no need for a porting note.

File: Mathlib/Data/Polynomial/Module.lean [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+variable (R M : Type _) [CommRing R] [AddCommGroup M] [Module R M] (I : Ideal R)
+
+--porting note: stated instead of deriving
+noncomputable instance : Inhabited (PolynomialModule R M) := Finsupp.inhabited
+noncomputable instance : AddCommGroup (PolynomialModule R M) := Finsupp.addCommGroup
+
+-- porting note: commenting out omit
+-- omit R
+
```
Comment (semorrison) [inline Mathlib/Data/Polynomial/Module.lean:66]:
```suggestion
```

File: Mathlib/Data/Polynomial/Module.lean [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+@[semireducible]
+noncomputable instance polynomialModule : Module R[X] (PolynomialModule R M) :=
+  modulePolynomialOfEndo (Finsupp.lmapDomain _ _ Nat.succ)
+#align polynomial_module.polynomial_module PolynomialModule.polynomialModule
+
+instance (M : Type u) [AddCommGroup M] [Module R M] [Module S M] [IsScalarTower S R M] :
+    IsScalarTower S R (PolynomialModule R M) :=
+  Finsupp.isScalarTower _ _
+
+instance is_scalar_tower' (M : Type u) [AddCommGroup M] [Module R M] [Module S M]
```
Comment (semorrison) [inline Mathlib/Data/Polynomial/Module.lean:124]:
```suggestion
instance isScalarTower' (M : Type u) [AddCommGroup M] [Module R M] [Module S M]
```

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**